File: 2022/P02/problem2a.py
data = []
with open("problem2.txt", 'r') as f:
    data = f.read().strip('\n').split('\n')

def f(char):
    if char == 'A' or char == 'X':
        return 1
    if char == 'B' or char == 'Y':
        return 2
    if char == 'C' or char == 'Z':
        return 3

score = 0
for p in data:
    chars = p.split(' ')
    nums = [f(x) for x in chars]
    round_score = 0
    if nums[0] == nums[1]:
        round_score += 3
    elif (nums[1] - nums[0] + 2) % 3 == 0:
        round_score += 6
    score += nums[1] + round_score

print(score)

File: 2023/P02/problem2a.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("problem2.txt")
        .expect("Failed to read file");
    let data = data.trim()
        .split('\n');
    let mut sum = 0;
    'main: for (i, el) in data.enumerate() {
        let games = el.trim().split(": ").collect::<Vec<_>>()[1];
        for g in games.split("; ") {
            let vals = g.split(", ");
            for v_col in vals {
                let vec = v_col.split(" ").collect::<Vec<_>>();
                let v = vec[0];
                let col = vec[1]; 
                match v.parse::<u32>() {
                    Ok(num) => {
                        match col {
                            "red" => {
                                if num > 12 {
                                    continue 'main;
                                }
                            },
                            "green" => {
                                if num > 13 {
                                    continue 'main;
                                }
                            },
                            "blue" => {
                                if num > 14 {
                                    continue 'main;
                                }
                            },
                            _ => {}
                        }
                    },
                    _ => {}
                }
            }
        }
        sum += i + 1;
    }
    println!("{sum}");
}
